assign a type assertion of func

The following program `sample.go` triggers a panic:

```go
package main

import (    
    "reflect"
    "time"
)

func main() {
    t := time.Date(2009, time.November, 10, 23, 4, 5, 0, time.UTC)
    v := reflect.ValueOf(t.String)
    f := v.Interface().(func() string)
    println(f())
}
```

Expected result:
```console
$ go run ./sample.go
2009-11-10 23:04:05 +0000 UTC
```

Got:
```console
$ yaegi ./sample.go
~/sample.go:9:7: panic
reflect.Set: value of type func() string is not assignable to type *interp.node
goroutine 1 [running]:
runtime/debug.Stack(0x1, 0xc00039e000, 0x40)
        /home/marc/sdk/go1.14.1/src/runtime/debug/stack.go:24 +0x9d
github.com/containous/yaegi/interp.(*Interpreter).Eval.func1(0xc0001bbe10)
        /home/marc/src/yaegi/interp/interp.go:320 +0xbc
panic(0xc93360, 0xc000286a90)
        /home/marc/sdk/go1.14.1/src/runtime/panic.go:967 +0x166
github.com/containous/yaegi/interp.runCfg.func1(0xc0003700a0, 0xc000381000)
        /home/marc/src/yaegi/interp/run.go:113 +0x20c
panic(0xc93360, 0xc000286a90)
        /home/marc/sdk/go1.14.1/src/runtime/panic.go:967 +0x166
reflect.Value.assignTo(0xc901a0, 0xc0002847c0, 0x13, 0xdc49aa, 0xb, 0xd6a7e0, 0x0, 0x665e1f, 0xc901a0, 0xc0002847c0)
        /home/marc/sdk/go1.14.1/src/reflect/value.go:2403 +0x426
reflect.Value.Set(0xd6a7e0, 0xc00028a250, 0x196, 0xc901a0, 0xc0002847c0, 0x13)
        /home/marc/sdk/go1.14.1/src/reflect/value.go:1532 +0xbd
github.com/containous/yaegi/interp.typeAssert.func1(0xc0003700a0, 0xc000288a60)
        /home/marc/src/yaegi/interp/run.go:167 +0xde
github.com/containous/yaegi/interp.runCfg(0xc000381000, 0xc0003700a0)
        /home/marc/src/yaegi/interp/run.go:119 +0x6d
github.com/containous/yaegi/interp.(*Interpreter).run(0xc00036a000, 0xc000380700, 0xc000370000)
        /home/marc/src/yaegi/interp/run.go:97 +0x239
github.com/containous/yaegi/interp.(*Interpreter).Eval(0xc00036a000, 0xc00037a000, 0xe5, 0x0, 0x0, 0x0, 0x0, 0x0)
        /home/marc/src/yaegi/interp/interp.go:388 +0x41f
main.main()
        /home/marc/src/yaegi/cmd/yaegi/yaegi.go:94 +0x548
```

--- a/_test/type21.go
+++ b/_test/type21.go
@@ -1,0 +1,16 @@
+package main
+
+import (
+	"reflect"
+	"time"
+)
+
+func main() {
+	t := time.Date(2009, time.November, 10, 23, 4, 5, 0, time.UTC)
+	v := reflect.ValueOf(t.String)
+	f := v.Interface().(func() string)
+	println(f())
+}
+
+// Output:
+// 2009-11-10 23:04:05 +0000 UTC

--- a/interp/cfg.go
+++ b/interp/cfg.go
@@ -1503,7 +1503,12 @@ func (interp *Interpreter) cfg(root *node, pkgID string) ([]*node, error) {
 					}
 				}
 				if n.anc.action != aAssignX {
-					n.typ = n.child[1].typ
+					if n.child[0].typ.cat == valueT {
+						// Avoid special wrapping of interfaces and func types.
+						n.typ = &itype{cat: valueT, rtype: n.child[1].typ.TypeOf()}
+					} else {
+						n.typ = n.child[1].typ
+					}
 					n.findex = sc.add(n.typ)
 				}
 			} else {

